### ISSUE TITLE
Add rotation limiting to slider component

### ISSUE THREAD

**Issue by mralext20** [Frooxius/NeosPublic#905]
Right now, the slider has a single on / off for rotational motion, it would be useful for there to be the ability to select a plane (by choosing 1,0for directions that the slider can rotate in. Ex. 

X:1 y:0, z:0 would only allow rotation on the x axis

this could be made more complex by having range for rotation, but I doubt that would make sence as quaternions get convert back and forth with euler angles.

**Comment by Frooxius** [comment]
I'm sorry, this won't get added to Slider as it's responsibility is a limited translation movement, but not much else. Rotation limiting is already handled by the Joint component and this would duplicate that behavior.

Are you looking to use both movement and limited rotations at once? I can instead add mechanisms to be able to use both components at the same time, Slider handling position and Joint handling rotation.

**Comment by mralext20** [comment]
Yes, that is the goal. Intended use is on a chair, so you can pull the chair out from under a table, then rotate it on the floor

( like a chair with spinny wheels)
